frontend_lib/src/mention.js

```diff
@@ -17,111 +17,101 @@ export const GROUP_MENTION_LIST = [
 
 export const GROUP_MENTION_TRANSLATION_LIST = ['all', 'tous', 'todos']
 
-const depthFirstSearchAndMentionAnalysis = childNodesList => {
-  const childNodesListCopy = [...childNodesList]
-  let i = 0
-
-  childNodesListCopy.forEach((node) => {
-    const value = node.nodeValue
-
-    if (node.nodeName === '#text' && value.includes('@')) {
-      const mentionsInThisNode = value.split(/\s/).filter(token => MENTION_REGEX.test(token))
-
-      if (mentionsInThisNode.length > 0) {
-        let mentionIndex = 0
-        let lastMentionIndex = 0
-        const fragment = document.createDocumentFragment()
-        let htmlTagCounter = 0
-
-        mentionsInThisNode.forEach((mention, i) => {
-          mentionIndex = value.indexOf(mention, lastMentionIndex)
-
-          const mentionWithSpan = document.createElement(MENTION_TAG_NAME)
-          mentionWithSpan.className = MENTION_CLASS
-          mentionWithSpan.id = `${MENTION_ID_PREFIX}${uuidv4()}`
-          mentionWithSpan.textContent = mention
-
-          if (mentionIndex !== 0) {
-            htmlTagCounter++
-            fragment.appendChild(document.createTextNode(value.substring(lastMentionIndex, mentionIndex)))
-          }
-
-          fragment.appendChild(mentionWithSpan)
-          htmlTagCounter++
-
-          if (mentionsInThisNode.length - 1 === i) {
-            htmlTagCounter++
-            fragment.appendChild(document.createTextNode(value.substring(mentionIndex + mention.length)))
-          }
-
-          lastMentionIndex = mentionIndex + mention.length - 1
-        })
-        childNodesList[i].replaceWith(fragment)
-        i = i + htmlTagCounter
-      } else i++
-    } else {
-      if (!(node.nodeName.toLowerCase() === MENTION_TAG_NAME && node.id.startsWith(MENTION_ID_PREFIX))) depthFirstSearchAndMentionAnalysis(node.childNodes)
-      i++
-    }
-  })
+const wrapMentionsInText = (doc, text) => {
```

Comment:
```suggestion
const wrapMentionsOfText = (text, doc) => {
```
`wrapMentionsInText` made me think that the function was wrapping the mentions found in the doc using the text parameter to wrap. Since "wrap in *something*" indicates the *something* is used to wrap.

Reply:
I renamed this, added a comment saying what it does, swapped the arguments and also swapped them in wrapMentionsInSpanTags for consistency.